web/html/src/manager/proxy/container-config.tsx

```diff
@@ -0,0 +1,377 @@
+        proxyAdminEmail: model.proxyAdminEmail,
+        sslMode: model.sslMode,
+      };
+
+      const cnamesData = Object.fromEntries(Object.entries(model).filter(([key]) => key.startsWith("cnames")));
+      const extraData =
+        model.sslMode === SSLMode.CreateSSL
+          ? Object.assign(
+              {},
+              {
+                caPassword: model.caPassword,
+                country: model.country,
+                state: model.state,
+                city: model.city,
+                org: model.org,
+                orgUnit: model.orgUnit,
+                sslEmail: model.sslEmail,
+              },
+              cnamesData
+            )
+          : {};
+      const formData = unflattenModel(Object.assign({}, commonData, extraData, ...values));
+      Network.post("/rhn/manager/api/proxy/container-config", formData).then(
+        (data) => {
+          setSuccess(data.success);
+          setMessages([]);
+          setLoading(false);
+          window.location.assign("/rhn/manager/api/proxy/container-config/" + data);
+        },
+        (xhr) => {
+          try {
+            setSuccess(false);
+            setMessages([JSON.parse(xhr.responseText)]);
+            setLoading(false);
+          } catch (err) {
+            const errMessages =
+              xhr.status === 0
+                ? t("Request interrupted or invalid response received from the server.")
+                : Network.errorMessageByStatus(xhr.status)[0];
+            setSuccess(false);
+            setMessages([errMessages]);
+            setLoading(false);
+          }
+        }
+      );
+    });
+  };
+
+  const clearFields = () => {
+    setModel(initialModel);
+  };
+
+  const onValidate = (isValidated: boolean) => {
+    setIsValidated(isValidated);
+  };
+
+  const onChange = (newModel: any) => {
+    setModel(Object.assign({}, newModel));
+  };
+
+  const onAddField = (fieldName: string) => {
+    return (index: number) => setModel(Object.assign({}, model, { [fieldName + index]: "" }));
+  };
+
+  const onRemoveField = (fieldName: string) => {
+    return (index: number) => {
+      const newModel = { ...model };
+      delete newModel[`${fieldName}${index}`];
+      setModel(newModel);
+    };
+  };
+
+  return (
+    <TopPanel
+      title={t("Container Based Proxy Configuration")}
+      icon="fa fa-cogs"
+      helpUrl="reference/proxy/container-based-config.html"
+    >
+      <p>
+        {t(
+          "You can generate a set of configuration files and certificates in order to register and run a container-based proxy. Once the following form is filled out and submitted you will get a .zip archive to download."
+        )}
+      </p>
+      {ContainerConfigMessages(success, messages, loading)}
+      <Form
+        className="form-horizontal"
+        model={model}
+        onValidate={onValidate}
+        onChange={onChange}
+        onSubmit={onSubmit}
+        title={t("Container Based Proxy Configuration")}
```

Comment:
One question about casing here, shouldn't the Pascal Case here be handled by styling, rather than manually? I don't know how we do it elsewhere though, if this is how we do it everywhere, feel free to disregard.

Reply:
It's a good point you are bringing it up. It is manually handled at the moment AFAIK, either when it comes from a JSP or from the React one. I don't even like that much a Pascal Case for all the page titles, but that's how it is done and for consistency reason I'd better keep it as-is now. I hope you don't feel hurt by that :smiley: , sorry

Reply:
That is very fine by me, simply wanted to point it out and ask what the rationale was. Your approach makes sense to me. 👍